### PR TITLE
Setup `aormsby/Fork-Sync-With-Upstream-action` to sync from upstream automatically

### PR DESCRIPTION
**Pull request by arunsathiya**
Not sure if this is something you'd want but figured I'd start a PR to create this discussion. I am assuming you merge all changes from upstream and then remove what you don't want?

Commits:

- Setup upstream repo sync for everyday at 7am: main branch to main branch
- Add note about GITHUB_TOKEN secret
- Delete upstream_repo_access_token as it is only for private upstream repositories

### CODE REVIEW

**Comment by sharunkumar** [comment]
I only merge when there is a new release tag upstream. is there a way to configure that? rather than for un-released main changes upstream

**Review by sharunkumar — CHANGES_REQUESTED** [review]
I only merge when there is a new release tag upstream. is there a way to configure that? rather than for un-released main changes upstream. also would prefer if I can change the commit message of the merge as "Merge Release xxx"

File: .github/workflows/upstream_sync.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        test_mode: false
+      
+    # Step 3: Display a sample message based on the sync output var 'has_new_commits'
+    - name: New commits found
+      if: steps.sync.outputs.has_new_commits == 'true'
+      run: echo "New commits were found to sync."
+    
+    - name: No new commits
+      if: steps.sync.outputs.has_new_commits == 'false'
+      run: echo "There were no new commits."
```
Comment (sharunkumar) [inline .github/workflows/upstream_sync.yml:43]: Interesting, I could use this logic for my other forks:

https://github.com/sharunkumar/Summer2024-Internships

https://github.com/sharunkumar/New-Grad-Positions

Comment (arunsathiya) [reply .github/workflows/upstream_sync.yml:43]: I don't think it's a GitHub workflows-wide output, rather an output variable specific to the `Fork-Sync-With-Upstream-action` action:

https://github.com/aormsby/Fork-Sync-With-Upstream-action/blob/f0889e09db4cb5657459c9256a9fb62a740d4f93/run/get_updates.sh#L49-L51

**Comment by arunsathiya** [comment]
It does seem possible to run workflows only when tags are created:

- https://github.com/orgs/community/discussions/25302

But, I cannot spot anything to indicate running the workflow only when a tag is created upstream. Still researching what's possible, but I am starting to think that this must be a feature request for the `aormsby/Fork-Sync-With-Upstream-action` action.

On a different note, is it intentional that your tags and upstream tags are not in sync?

- https://github.com/aeharding/voyager/tags
- https://github.com/sharunkumar/voyager/tags

File: .github/workflows/upstream_sync.yml (hunk 1 above)
Comment (sharunkumar) [reply .github/workflows/upstream_sync.yml:43]: Yeah I got that, just saying that it could be helpful in those repos

**Comment by arunsathiya** [comment]
I'll move this to draft until we know for sure:

- Whether workflows can be run for upstream tags: https://github.com/orgs/community/discussions/25302#discussioncomment-7258370
- If not, until that feature is available in the `aormsby/Fork-Sync-With-Upstream-action` action: https://github.com/aormsby/Fork-Sync-With-Upstream-action/issues/71

**Comment by sharunkumar** [comment]
@arunsathiya current release https://github.com/aeharding/voyager/releases/tag/1.14.0 is an example of heavy merge conflicts. this can't be automated. so I'll have to close this PR for now as not planned

**Comment by sharunkumar** [comment]
> On a different note, is it intentional that your tags and upstream tags are not in sync?

I guess I have not pushed the tags to my repo

**Comment by sharunkumar** [comment]
also looking around on the internet, its possible to merge the latest tag from upstream: https://stackoverflow.com/a/57877266

Made a gist for reference
https://gist.github.com/sharunkumar/4f372889fd0feec01a96fa1f13b4708f

**Pull request closed without merging**